src/server/index.tsx

```diff
@@ -170,25 +174,34 @@ server.get("/*", async (req, res) => {
         };
 
         if (activeRoute?.fetchInitialData) {
-          routeData.push(
-            ...(await Promise.all([
-              ...activeRoute.fetchInitialData(initialFetchReq),
-            ]))
+          const routeDataKeysAndVals = await Promise.all(
+            Object.entries(activeRoute.fetchInitialData(initialFetchReq)).map(
```

Comment:
All the fetchInitialData got turned async, so I really hope this is working as intended.

Reply:
It wasn't, but it is now.